src/index.test.ts

```diff
@@ -244,6 +244,49 @@ test('api on response', async () => {
   server.close();
 });
 
+test('api on pre build modifies auth token', async () => {
+  const server = http
+    .createServer(({ headers }, res) => {
+      res.writeHead(200, { 'Content-Type': 'application/json' });
+      res.end(
+        JSON.stringify({
+          headers,
+        }),
+      );
+    })
+    .listen(0);
+
+  const { port } = server.address() as { port: number };
+
+  const auth = {
+    token: 'abc',
+  };
+
+  const test = api(`http://localhost:${port}`)
+    .withBearerToken(auth)
+    .onPreBuild(() => {
+      auth.token = 'def';
```

Comment:
Interesting - is this the motivating use case?

Reply:
Yup - better explained in this PR's description, but essentially want a way to automatically refresh auth tokens (eg. when server side talking to a third party API) before requests are made.